module/multicast.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""!
+    ____  ____  ______       __      __       __       _____
+   / __ )/ __ \/ ___/ |     / /___ _/ /______/ /_     |__  /
+  / __  / / / /\__ \| | /| / / __ `/ __/ ___/ __ \     /_ <
+ / /_/ / /_/ /___/ /| |/ |/ / /_/ / /_/ /__/ / / /   ___/ /
+/_____/\____//____/ |__/|__/\__,_/\__/\___/_/ /_/   /____/
+                German BOS Information Script
+                     by Bastian Schroll
+
+@file:        multicast.py
+@date:        02.05.2021
+@author:      Thierry Fredrich
+@description: Implements multicast alarms
+
+structure of a multicast alarm 
+# 1 ) network delimiter without text 
+# 2 ) alarm ric without text 
+# 3 ) text ric with message 
+# 4 ) network delimiter ric message <BS>
+"""
+import logging
+from module.moduleBase import ModuleBase
+
+
+logging.debug("- %s loaded", __name__)
+
+
+class BoswatchModule(ModuleBase):
+    """!Description of the Module"""
+    # initializing with empty list
+    
+    textRics = []
+    delimiterRic = 42
+    delimiterSubric = 42
+    ignoreTime = 42
+    ### to be cleared
+    receivedAlarmRic = 42
+    receivedAlarmSubric = 42
+    bufferListFormingAMulticastAlarm = []
+    initialDelimiterReceived = False
+    alarmReceived = False
+    textRicReceived = False
+    textRicMessage = ''
+    def initStorage(self):
+        self.receivedAlarmRic = 42
+        self.receivedAlarmSubric = 42
+        self.bufferListFormingAMulticastAlarm = []
+        self.initialDelimiterReceived = False
+        self.alarmReceived = False
+        self.textRicReceived = False
+        self.textRicMessage = ''
+    
+    def __init__(self, config):
+        """!Do not change anything here!"""
+        super().__init__(__name__, config)  # you can access the config class on 'self.config'
+        logging.debug("starting multicast module")
+        logging.debug("multicastAlarm_delimiterRic is: %i" % self.config.get("multicastAlarm_delimiterRic"))
+        logging.debug("multicastAlarm_delimiterSubric is: %i" % self.config.get("multicastAlarm_delimiterSubric"))
+        logging.debug("multicastAlarm_ignore_time is: %i" % self.config.get("multicastAlarm_ignore_time"))
+        logging.debug("multicastAlarm_textRics is: %s" % self.config.get("multicastAlarm_textRics"))
+        self.delimiterRic = int(self.config.get("multicastAlarm_delimiterRic"))
+        self.delimiterSubric = int(self.config.get("multicastAlarm_delimiterSubric"))
+        self.ignoreTime = int(self.config.get("multicastAlarm_ignore_time"))
+        for aTextRic in self.config.get("multicastAlarm_textRics").split(','):
+            self.textRics.append(int(aTextRic))
+        self.initStorage()

```

Comment:
This should be done in the `onLoad()` method below.
The Init must not be changed.

Reply:
I see. I will change that accordingly.